File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        vector<int> vec1(26, 0);
        vector<int> vec2(26, 0);
        for (char c : s1) {
            vec1[c - 'a']++;
        }
        for (char c : s2) {
            vec2[c - 'a']++;
        }
        if (vec1 == vec2) {
            return true;
        }
        return false;
    }
};

File: LeetCode/1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        if s1 == s2:
            return True
        left, right = 0, len(s1) - 1
        while s1[left] == s2[left]:
            left += 1
        while s1[right] == s2[right]:
            right -= 1
        s2_list = list(s2)        
        s2_list[left], s2_list[right] = s2_list[right], s2_list[left]
        return s1 == ''.join(s2_list)